src/transformers/trainer.py

```diff
@@ -4580,9 +4587,12 @@ def prediction_loop(
             if self.args.batch_eval_metrics:
                 if self.compute_metrics is not None and preds_host is not None and labels_host is not None:
                     is_last_step = self.accelerator.gradient_state.end_of_dataloader
-                    if args.include_inputs_for_metrics:
+                    if args.include_inputs_for_metrics or args.include_loss_for_metrics:
+                        batch_kwargs = {}
+                        batch_kwargs["losses"] = losses_host if args.include_loss_for_metrics else None
+                        batch_kwargs["inputs"] = inputs_host if args.include_inputs_for_metrics else None
                         metrics = self.compute_metrics(
-                            EvalPrediction(predictions=preds_host, label_ids=labels_host, inputs=inputs_host),
+                            EvalPrediction(predictions=preds_host, label_ids=labels_host, **batch_kwargs),
                             compute_result=is_last_step,
                         )
                     else:
```

Comment:
We can further simplify these ! let's remove the if else and call metrics = self.compute_metrics only once.